### PR TITLE
[Magiclysm] Fun with Flags, spell edition

### PR DESCRIPTION
**Pull request by Standing-Storm**
#### Summary
Mods "[Magiclysm] Fun with Flags, spell edition"

#### Purpose of change

Add more spells specifically themed around manipulating item flags

#### Describe the solution

Add the following spells:

ANIMIST

- Abjure Spirits of Rot (Diff 5): Banish disease spirits from food or drink the mage is carrying, making it last much longer before going bad. The actual principles by which this spell worked were not discovered until the 20th century, and by that point the name had stuck (applies the IRRADIATED flag to food items. An Animist spell because Radiation Mage is an Animist attunement). Component: 15 wild herbs and 20 sulfur.

EARTHSHAPER:

- Fault-Mending Chant (Diff 4): While holding an item made of natural metal or stone, whisper the words of the chant and smooth out its imperfections, repairing it. Component: soil, stone, and a small amount of metal (copper, silver, gold, iron, lead, or platinum)

(Natural metal [i.e. no steel or alloys] required to make Magiclysm's "technology is better than magic" world make sense, otherwise every factory would just have a bunch of wizards on staff to rub dirt on broken machines. Lore-wise I could say, "Oh, nothing with complex moving parts is eligible" but I don't believe there's an easy way to enforce that in-game)

KELVINIST:

- Baking Hands (Diff 1): Unfreeze and heat up a single item of food or drink--not enough to cook it, but enough to make it nice and hot before eating it. Component: 1 charcoal or 1 coal

MAGUS:

- Diamond Imbuement (Diff 8): Infuse a weapon with the metaphysical essence of a diamond's hardiness and cutting power, making it do more damage and be more resilient.  However, like a diamond, any cracks or flaws cannot be fixed without destroying it. (applies the DURABLE_MELEE and DIAMOND flags, but also the NO_REPAIR flag). Component: 5 diamonds

TECHNOMANCER:

- Power Linkage (Diff 6): Modify a tool to draw power wirelessly from nearby sources, allowing it to use a UPS [or compatible bionic power connection] to recharge itself. Pairs well with the spell Power Supply. Components: Crude lamp oil, kerosene, gasoline, or diesel

I also changed the the name of the Animist Spirit-Walking spell to Body of Spirit, because I forgot there was already an Animist spell called "Spirit Walk."   Furthermore, I modified Power Supply because a lot of UPS items that don't take batteries require a minimum charge to start, so giving the power supply a maximum storage of 1 makes it not that useful even if it does regenerate energy.

More to come.

#### Testing

Spells work properly.

#### Additional context

Caught moves ->  second during testing, the new Power Supply will not last 100x longer than it's supposed to.
More spells need components, honestly, it's a neat system, helps balance magic, and ties into the classic fun part of CDDA gameplay--rooting through the ruins of the old world.

### CODE REVIEW

File: data/mods/Magiclysm/Spells/kelvinist.json [hunk 1]
```diff
@@ -707,5 +707,38 @@
     "sound_type": "combat",
     "sound_id": "fire_spell",
     "sound_description": "a crackle!"
+  },
+  {
+    "id": "kelvinist_heat_food",
+    "type": "SPELL",
+    "name": "Baking Hands",
+    "description": "With the tiniest amount of eat, unfreeze an item of food or drink and heat it up to a comfortable temperature.",
```
Comment (TheShadowFerret) [inline data/mods/Magiclysm/Spells/kelvinist.json:715]:
```suggestion
    "description": "With the tiniest amount of heat, unfreeze an item of food or drink and heat it up to a comfortable temperature.",
```

File: data/mods/Magiclysm/items/spell_scrolls.json [hunk 2]
```diff
@@ -1986,5 +1986,32 @@
...
     "description": "Gain an intuitive understanding of computers and electronics, allowing an easier time repairing or using them.",
     "use_action": { "type": "learn_spell", "spells": [ "technomancer_gain_electronics_computer_spell" ] }
+  },
+  {
+    "type": "BOOK",
+    "copy-from": "spell_scroll",
+    "id": "spell_scroll_kelvinist_heat_food",
+    "//": "Kelvinist spell",
+    "name": { "str": "Scroll of Baking Hands", "str_pl": "Scrolls of Baking Hands" },
+    "description": "Warm up your food using pure mana.",
```
Comment (TheShadowFerret) [inline data/mods/Magiclysm/items/spell_scrolls.json:1996]: +Charcoal isn't pure mana 🤔

File: data/mods/Magiclysm/Spells/magus.json [hunk 3]
```diff
@@ -644,5 +644,39 @@
     "energy_source": "MANA",
     "base_energy_cost": 350,
     "base_casting_time": 300
+  },
+  {
+    "id": "magus_diamond_imbuement",
+    "type": "SPELL",
+    "name": "Diamond Imbuement",
+    "description": "Magically infuse a melee weapon with the hardness and resilience of a diamond as well as its industrial cutting capability, making it ",
```
Comment (TheShadowFerret) [inline data/mods/Magiclysm/Spells/magus.json:652]:
```suggestion
    "description": "Magically infuse a melee weapon with the hardness and resilience of a diamond as well as its industrial cutting capability, making it supremely sharp and durable.",
```

**Review by TheShadowFerret — COMMENTED** [review]
~~How can **baking** hands not bake reeeee~~

File: data/mods/Magiclysm/items/spell_scrolls.json (hunk 2 above)
Comment (Standing-Storm) [reply data/mods/Magiclysm/items/spell_scrolls.json:1996]: The charcoal fuels the spell, not the heating--you're not actually burning the charcoal, just using its sympathetic connection to cooking.

**Comment by Cupidatis** [comment]
Is Diamond Imbuement permanent?

**Comment by Standing-Storm** [comment]
> Is Diamond Imbuement permanent?

Technically yes, but weapons with the DIAMOND flag cannot be repaired.

**Comment by Cupidatis** [comment]
> > Is Diamond Imbuement permanent?
> 
> Technically yes, but weapons with the DIAMOND flag cannot be repaired.

Oh, ok.

**Pull request merged**